Lib/asyncio/stack.py

```diff
@@ -0,0 +1,231 @@
+"""Introspection utils for tasks call stacks."""
+
+import sys
+import types
+import typing
+
+from . import events
+from . import futures
+from . import tasks
+
+__all__ = (
+    'capture_call_graph',
+    'print_call_graph',
+    'FrameCallGraphEntry',
+    'FutureCallGraph',
+)
+
+# Sadly, we can't re-use the traceback's module datastructures as those
+# are tailored for error reporting, whereas we need to represent an
+# async call stack.
+#
+# Going with pretty verbose names as we'd like to export them to the
+# top level asyncio namespace, and want to avoid future name clashes.
+
+
+class FrameCallGraphEntry(typing.NamedTuple):
+    frame: types.FrameType
+
+
+class FutureCallGraph(typing.NamedTuple):
+    future: futures.Future
+    call_stack: list[FrameCallGraphEntry]
+    awaited_by: list[FutureCallGraph]
+
+
+def _build_stack_for_future(future: any) -> FutureCallGraph:
```

Comment:
Remove the type annotations, we don't have it for any other place in asyncio, also using any is wrong is as well

Reply:
Here I'm -1. I think we should actually start using type annotations, we already do that in other places of the standard library.

Annotations improve readability and maintainability significantly. At least new code can benefit from them.

Reply:
Okay, then please fix the type annotations, any is not a valid annotation

Reply:
Yeah, @ambv just pointed that out to me too :) My bad, absolutely, they will be fixed.

To drive my point though, Kumar, I think let's not hold back on annotations, I think asyncio code would benefit from them.